lib/utils.pm

```diff
@@ -588,6 +588,7 @@ sub zypper_call {
     my $timeout = $args{timeout} || 700;
     my $log = $args{log};
     my $dumb_term = $args{dumb_term} // is_serial_terminal;
+    my $retry = $args{retry} || 5;
```

Comment:
`my $retry = $args{retry} // 5;` As `0` is a valid value IMO in case you simply want to have 0 retries.

Reply:
0 reties means No-run at all, being the command `for 1 .. 0`: isn't it?
Also, the symbol `//` is used when `undefined`, so the $retry can even be empty and provide error

Reply:
IMHO, the loop type is not correct. It should  always run at least once. If there is undef the retry will be the default value. In this implementation we cannot disable retry.

Reply:
I left the previous old existing setting for default: 
if `$retry` is empty, then 5 assigned, as before it worked.

But also retry=0, here becomes 5

Reply:
In last update: `undef`=5; `assigned but '' or 0`=1; `>0`=$retry.

Reply:
In last update I unified the settings, adding a description:
`# (retry >=0 or ='': unchanged; undef: 5) result ='' or 0: 1`

Reply:
The base of the problem is following:

- user wants to run a `X_retry` so it means it will always run at least once
- number of retries can be parametrized, that is nice, but it should lead to `1 + retries` runs

In this implementation the caller would think that there are 5 retries at least, but there are only 4 in fact.

Reply:
IMO this would simply the code and it would be also cleaner, straightforward and maybe more safe

Reply:
It is a matter of definitions: _retries after `first` attempt failed_ or _total runs while failing_ (we could rename variable).
In this implementation: 1 means 1 run only total (no retry); 2 means 1 retry if failed or 2 total runs whil failing, etc...
To well-use our routines we have the heading routine descipition: the user should clarify all logic and usage there: I will update it too, when behavior clarified.

Reply:
I have a feeling that this PR is diverging into all possible spaces at once. I would be happy to see that `zypper_call` is refactored/updated/you name it in a separate PR, slow track that all users can drop their insights and thoughts of to make the code better. I do not see any strong relation between `ppc` and `zypper_call` that is required at the moment. It would be beneficial for everyone if `zypper_call` improvements would be separated and that you can improve that code (which would be welcomed) in a different PR :)

Reply:
Yes I already mentioned it before in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20383#issuecomment-2407440264.
A weak correlation and motivation that led me to this change is the introduction of sure failing check on sle-micro CA check: original loop is 5 retry, I attempting to reduce it to 3 at most.
But if it is a problem, I will revert this change and move it to a different next PR.

Reply:
The problem is that cert URL seems to be wrongly composed. If that is fixed, all the stuff should run smoothly as it runs well for other archs